### PR TITLE
feat: Adjusted telemetry 'deploymentes' suppression

### LINKED ISSUE #1344: [AVM Question/Feedback]: Supress 'no-deployments' resource warnings
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

Currently, when using an AVM module, users see a warnings for the `Microsoft.Resources/deployments` deployment that we use for the telemetry tracking. While a best-practice, this design was choises in this case for a purpose.
We should investigate if we can either supress it, or, if not, change the implementation to not cause it.

> Warning no-deployments-resources: Resource 'avmTelemetry' of type 'Microsoft.Resources/deployments@2023-07-01' should instead be declared as a Bicep module

Mind you: Supressing it in the `bicepconfig.json` file will not be enough.

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added deployments suppression to module instead of just bicepconfig to also apply to anybody using our modules without a likewise configure bicepconfig locally

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.analysis-services.server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml)     |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: avm/res/api-management/service/main.bicep
Comment (eriqua) [inline avm/res/api-management/service/main.bicep:133]: I'd kindly ask to merge #1481 first, to simplify the review of this PR and correctly track the modules still missing v2 formatting

Comment (AlexanderSehr) [reply avm/res/api-management/service/main.bicep]: done

File: avm/res/app/managed-environment/main.bicep
Comment (eriqua) [inline avm/res/app/managed-environment/main.bicep:76]: This and other formatting changes should be taken care by the related open PRs

**Review by jeetgarg — APPROVED** [review]

File: avm/res/app/managed-environment/main.bicep
Comment (AlexanderSehr) [reply avm/res/app/managed-environment/main.bicep:76]: ... as those PRs  should also already apply the formatting (since we removed the V1 enforcement) I don't expect any conflicts if we were to merge this one in first. 💪 
So, is the idea of the comment to block this one regardless until the x other PRs are merged ? Or just to undo the reformatting (which is automatically applied), so I'd need to add the new feature via Notepad.

Comment (eriqua) [reply avm/res/app/managed-environment/main.bicep:76]: I'm a bit lost. Didn't we remove v2 formatting from the bulk PR for any module having a PR already open? What is the difference if we add the formatting in this PR?

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/main.bicep:76]: Nah, enforcing V1 one doesn't unformat 😄 . I need to remove the changes by setting the files back to their state in main, then enforce V1, apply the change anew, un-enforce V1 and hope for the best 😄

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/main.bicep:76]: The difference to the previous PR is that we now have removed the V1 formatter enforcement. So 'IF' any of the open PRs stay up to date with upstream, they should all apply the reformatting on their end automatically (if I'm not mistaken).

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/main.bicep:76]: Regardless - you answered my question. I'll try to look into undoing the changes later this week as it's too much manual work for the time of day 😄

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/main.bicep:76]: Nono absolutely. 😄 just talking about this PR.
If their PRs are merged before this one I may anyways run into a merge conflict. I don't think there is necessarily a good solution unless both use the same formatting.

Comment (eriqua) [reply avm/res/app/managed-environment/main.bicep:76]: Resolving this as not relevant anymore.

**Comment by AlexanderSehr** [comment]
Hey @eriqua, will close and completely redo this PR as it has been going too long without approval, now being a pit of conflicts. More luck next time

**Pull request closed without merging**
